### PR TITLE
Upgrade to gradle 8.10

### LINKED ISSUE #13698: Upgrade to gradle 8.10
### Description

As in the title.

### Version and environment details

_No response_

### PR DESCRIPTION
**Pull request by dweiss**
Fixes #13698.

I can't claim I've tested everything but most things seem to work.

### CODE REVIEW

File: gradle/documentation/render-javadoc.gradle [hunk 1]
```diff
@@ -32,7 +32,7 @@ allprojects {
       missingdoclet "org.apache.lucene.tools:missing-doclet"
     }
 
-    ext {
+    project.ext {
```
Comment (ChrisHegarty) [inline gradle/documentation/render-javadoc.gradle:35]: Hmm.. it's odd that we have to do this, but I see that it's now necessary (why Gradle.. why!? ).  This is reason that my previous local attempt to upgrade failed. Good that you have a fix.

**Review by ChrisHegarty — APPROVED** [review]
I tested a number of different JDKs, 21, 22, and 23-ea, with various different test specific properties and env variables, and all worked successfully. LGTM.

**Review by uschindler — COMMENTED** [review]
looks good. I will report issues in the seldomy executed tasks (like regenerate)

**Comment by uschindler** [comment]
I also tested `gradlew :lucene:core:regenerate` and `gradlew eclipse`. Both work and the availability checks of toolkit autodownload works.

**Comment by dweiss** [comment]
> looks good. I will report issues in the seldomy executed tasks (like regenerate)

This is already sanity checked when you touch/ alter gradle files, along with other things.
https://github.com/apache/lucene/blob/main/.github/workflows/run-checks-gradle-upgrade.yml#L75-L86

Thanks for checking Eclipse, this is a bit manual.

**Pull request merged**